config/webpack/environment.js

```diff
@@ -26,6 +26,8 @@ environment.splitChunks((config) => {
           __dirname,
           '../../app/javascript/shared/components',
         ),
+        react: 'preact/compat',
+        'react-dom': 'preact/compat',
```

Comment:
Is this so that modules with a hard dependency on React will instead import Preact's React-compatibility layer?

Reply:
Exactly. These [webpack aliases](https://webpack.js.org/configuration/resolve/#resolvealias) enable that.